### ISSUE TITLE
Fix failing dev deployment

### ISSUE THREAD

**Issue by atolopko-czi** [chanzuckerberg/single-cell-data-portal#3830]
See https://czi-sci.slack.com/archives/C023Q1APASK/p1671664365669329

Fix:

Timmy: We should build the app and generate all the artifacts before pushing, so the cluster nodes don't need to spend time building static pages and randomly fails deployment due to timeout 

NOTE: We need to make sure the build step has access to `COMMIT_SHA` aka `HAPPY_COMMIT`, so the build id is consistent across all FE nodes

Tutorial:

https://dev.to/thakkaryash94/build-nextjs-application-using-github-workflow-and-docker-3foj

NOTES:
- [ ] Please reduce FE container resource from 8G to 2G in terraform, since we likely don't need more RAM than that for just running the app

**Comment by tihuan** [comment]
Jess comment:

https://czi-sci.slack.com/archives/C024P2G2KU1/p1668552020425519

**Comment by tihuan** [comment]
Jess is helping us with this now — the new approach is to build one image that can be used for dev/staging/prod and do text replacement per env, e.g., API_URL

**Comment by tihuan** [comment]
Jess is helping us out with this one! Her branch is:

`jgadling-test-runtime-vars`

Currently waiting on her feedback of local FE container not working for me

**Comment by ebezzi** [comment]
I tried the branch and the FE container seems to work fine, both locally and in rdev. Please re-test.

**Comment by tihuan** [comment]
New PR!

**Comment by tihuan** [comment]
For completeness, Emanuele was right that the branch did work, I just had to destroy all my container images locally and bump up RAM to 8GB for it to work on my machine 😆 👍

**Comment by ebezzi** [comment]
This should now be working. It will be in prod on Feb 24th.
